src/main/kotlin/com/openlattice/assembler/Assembler.kt

```diff
@@ -274,14 +276,51 @@ class Assembler(
 
     @Subscribe
     fun handleAddMembersToOrganization(event: MembersAddedToOrganizationEvent) {
+        // check if organization is initialized
+        ensureAssemblyInitialized(event.organizationId)
+
+        val authorizedPropertyTypesOfEntitySetsByNewMembers = event.newMembers.associateWith { principal ->
+            // we also grant select on entity sets, where no property type is authorized
+            val authorizedEntitySets = authorizationManager.getAuthorizedObjectsOfType(
+                    principal.principal,
+                    SecurableObjectType.EntitySet,
+                    EdmAuthorizationHelper.READ_PERMISSION
+            )
+            val authorizedPropertyTypeAcls = authorizationManager.getAuthorizedObjectsOfType(
+                    principal.principal,
+                    SecurableObjectType.PropertyTypeInEntitySet,
+                    EdmAuthorizationHelper.READ_PERMISSION
+            )
+
+            val allEntitySets = entitySets
+                    .getAll(authorizedEntitySets.map { it[0] }.collect(Collectors.toSet()))
+                    .filter { materializedEntitySets.containsKey(EntitySetAssemblyKey(it.key, event.organizationId)) }
```

Comment:
This is probably fine but if there's a way to create a `ContainsKeyPredicate` or a similar idea, it would definitely be preferable over filtering the results like this

Reply:
Doh, nice catch!